app/helpers/ticketing.py

```diff
@@ -303,6 +303,9 @@ def initiate_order_payment(form, paid_via=None):
         if order:
             user = DataGetter.get_or_create_user_by_email(email, form)
             order.user_id = user.id
+            if not (order.user.user_detail.firstname and order.user.user_detail.lastname):
```

Comment:
Logical error.. needs to be 
`if not order.user.user_detail.firstname and not order.user.user_detail.lastname`